webapp/src/main/java/com/box/l10n/mojito/quartz/multi/QuartzMultiSchedulerConfigurationProperties.java

```diff
@@ -0,0 +1,74 @@
+package com.box.l10n.mojito.quartz.multi;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.PostConstruct;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ConfigurationProperties("l10n.org.multi-quartz")
+@ConditionalOnProperty(name = "l10n.org.multi-quartz.enabled", havingValue = "true")
+public class QuartzMultiSchedulerConfigurationProperties {
+
+  Map<String, Map<String, String>> schedulers;
+
+  List<SchedulerConfigurationProperties> schedulerConfigurationProperties = new ArrayList<>();
+
+  public Map<String, Map<String, String>> getSchedulers() {
+    return schedulers;
+  }
+
+  public void setSchedulers(Map<String, Map<String, String>> schedulers) {
+    this.schedulers = schedulers;
+  }
+
+  public List<SchedulerConfigurationProperties> getSchedulerConfigurationProperties() {
+    return schedulerConfigurationProperties;
+  }
+
+  @PostConstruct
+  public void init() {
+    schedulerConfigurationProperties.addAll(extractSchedulerConfigs());
+  }
+
+  private List<SchedulerConfigurationProperties> extractSchedulerConfigs() {
+    List<SchedulerConfigurationProperties> schedulerProperties = new ArrayList<>();
+
+    for (Map.Entry<String, Map<String, String>> entry : schedulers.entrySet()) {
+      SchedulerConfigurationProperties config = new SchedulerConfigurationProperties();
+      config.setName(entry.getKey());
+      for (Map.Entry<String, String> quartzEntry : entry.getValue().entrySet()) {
+        config.getQuartz().put(quartzEntry.getKey().replace("quartz.", ""), quartzEntry.getValue());
```

Comment:
I don't think we need this type of remapping. I suspect you keep "quartz" in the key to have quartz property and potentially Mojito's specific configuration per scheduler which I think make sense. 

This can be mapped with a subclass like this

```
static class SchedulerConfigurationProperties {
    Map<String, String> quartz = new HashMap<>();
    int someOtherConfig;

    public Map<String, String> getQuartz() {
      return quartz;
    }

    public void setQuartz(Map<String, String> quartz) {
      this.quartz = quartz;
    }

    public int getSomeOtherConfig() {
      return someOtherConfig;
    }

    public void setSomeOtherConfig(int someOtherConfig) {
      this.someOtherConfig = someOtherConfig;
    }
  }
  ```
  and a map for the schedulers
  ```
  
  Map<String, SchedulerConfigurationProperties> schedulers;

  public Map<String, SchedulerConfigurationProperties> getSchedulers() {
    return schedulers;
  }

  public void setSchedulers(
      Map<String, SchedulerConfigurationProperties> schedulers) {
    this.schedulers = schedulers;
  }
```

Reply:
We're reflecting how the existing scheduler config works here, quartz configuration properties sit under the scheduler name and as part of the startup process get put into a `Properties` object with the prefix `org.quartz` appended to each of the config values, which then get read correctly by the internal quartz startup process.

I'll update the config classes accordingly, I wasn't aware that Spring would be magic enough to extract a dynamic map key with a custom properties object.